File: src/domain/valueObjects/PaymentStatus.ts
import type { ObjectValues } from "$domain/types";

export const PAYMENT_STATUS = {
  Canceled: "Canceled",
  Refunded: "Refunded",
  Pending: "Pending",
  Payed: "Payed",
  Completed: "Completed",
} as const;

export type PaymentStatus = ObjectValues<typeof PAYMENT_STATUS>;

File: src-tauri/src/domain/value_object/payment_status.rs
use serde::{Deserialize, Serialize};
use strum_macros::{EnumIter, EnumString};

#[derive(
    Clone, Debug, Deserialize, EnumString, EnumIter, Eq, PartialEq, Serialize, strum_macros::Display,
)]
pub enum PaymentStatus {
    Canceled,
    Refunded,
    Pending,
    Payed,
    Completed,
}
